File: src/main/java/no/novari/flyt/archive/gateway/template/model/Selectable.java
package no.novari.flyt.archive.gateway.template.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;

@Getter
@Builder
@EqualsAndHashCode
@Jacksonized
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Selectable {

    @NotBlank
    private final String displayName;

    @NotNull
    private final String value;

}

File: src/main/kotlin/no/novari/flyt/archive/gateway/template/model/Selectable.kt
package no.novari.flyt.archive.gateway.template.model

import com.fasterxml.jackson.annotation.JsonInclude
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Selectable(
    @field:NotBlank
    val displayName: String? = null,
    @field:NotNull
    val value: String? = null,
) {
    companion object {
        @JvmStatic
        fun builder() = Builder()
    }

    class Builder {
        private var displayName: String? = null
        private var value: String? = null

        fun displayName(displayName: String?) = apply { this.displayName = displayName }

        fun value(value: String?) = apply { this.value = value }

        fun build() = Selectable(displayName = displayName, value = value)
    }
}
